src/tgen-transport.h

```diff
@@ -7,31 +7,39 @@
 
 #include "tgen.h"
 
-typedef enum _TGenTransportProtocol {
-    TGEN_PROTOCOL_NONE, TGEN_PROTOCOL_TCP, TGEN_PROTOCOL_UDP,
-    TGEN_PROTOCOL_PIPE, TGEN_PROTOCOL_SOCKETPAIR,
-} TGenTransportProtocol;
-
 typedef struct _TGenTransport TGenTransport;
 
 typedef void (*TGenTransport_notifyBytesFunc)(gpointer data, gsize bytesRead, gsize bytesWritten);
 
-TGenTransport* tgentransport_newActive(TGenPeer* proxy, gchar* username, gchar* password, TGenPeer* peer,
-        TGenTransport_notifyBytesFunc notify, gpointer data, GDestroyNotify destructData);
-TGenTransport* tgentransport_newPassive(gint socketD, gint64 started, gint64 created, TGenPeer* peer,
-        TGenTransport_notifyBytesFunc notify, gpointer data, GDestroyNotify destructData);
+typedef struct _NotifyBytesCallback NotifyBytesCallback;
+struct _NotifyBytesCallback {
```

Comment:
General question - I've noticed adding a name for the struct type even when that name is never used. e.g. here is there a reason not to just do:
```
typedef struct {
  ...
} NotifyBytesCallback;
```
?

Reply:
GLib does it that way and I basically followed suit. Then I started doing it everywhere I write C, basically as a consistency thing. I don't feel strongly about the way I did it vs. your suggestion, but I feel stronger about being consistent.